draft-irtf-cfrg-vdaf.md

```diff
@@ -3149,22 +3146,41 @@ def encode_public_share(IdpfPoplar, correction_words):
 def decode_public_share(IdpfPoplar, encoded):
     l = floor((2*IdpfPoplar.BITS + 7) / 8)
     encoded_ctrl, encoded = encoded[:l], encoded[l:]
-    packed_ctrl = OS2IP(encoded_ctrl)
+    control_bits = unpack_bits(encoded_ctrl, 2 * IdpfPoplar.BITS)
     correction_words = []
     for level in range(IdpfPoplar.BITS):
         Field = IdpfPoplar.current_field(level)
-        ctrl_cw = (Field2(packed_ctrl & 1),
-                   Field2((packed_ctrl >> 1) & 1))
-        packed_ctrl >>= 2
+        ctrl_cw = (
+            control_bits[level * 2],
+            control_bits[level * 2 + 1],
+        )
         l = IdpfPoplar.Prg.SEED_SIZE
         seed_cw, encoded = encoded[:l], encoded[l:]
         l = Field.ENCODED_SIZE * IdpfPoplar.VALUE_LEN
         encoded_w_cw, encoded = encoded[:l], encoded[l:]
         w_cw = Field.decode_vec(encoded_w_cw)
         correction_words.append((seed_cw, ctrl_cw, w_cw))
-    if packed_ctrl != 0 or len(encoded) != 0:
+    leftover_bits = encoded_ctrl[-1] >> (
+        ((IdpfPoplar.BITS + 3) % 4 + 1) * 2
+    )
+    if leftover_bits != 0 or len(encoded) != 0:
         raise ERR_DECODE
     return correction_words
+
+def pack_bits(bits: Vec[Field2]) -> Bytes:
```

Comment:
I'd suggest moving these definitions to the preliminaries section and specifying them in English. Too much code runs the risk of making the spec less accessible. Since this method is relatively simple, a prose description of these methods would be sufficient in my opinion. (An implementer could refer to the reference code.)